src/validation/body-check.ts

```diff
@@ -0,0 +1,32 @@
+import { Context } from 'hono';
+import { z, ZodError } from 'zod';
+
+export const BodySchema = z.object({
+  id: z.number().optional(),
+  name: z
+    .string()
+    .min(1)
+    .max(50)
+    .regex(/^[A-Za-z ]+$/),
+  email: z.string().regex(/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}/i),
+  password: z
+    .string()
+    .min(6)
+    .max(20)
+    .regex(/^[A-Za-z0-9]+$/),
```

Comment:
1. Your passwords are too weak, let’s add some special symbols there. I’d like to see at least 10 more symbols 
2. Try to use `\w` instead of `A-Za-z0-9`